File: simulations-java/simulations/the-ramp/scala-src/edu/colorado/phet/therampscala/swing/MyCheckBox.scala
package edu.colorado.phet.therampscala.swing


import java.awt.event.{ActionListener, ActionEvent}
import scala.swing.CheckBox

class MyCheckBox(text: String, setter: Boolean => Unit, getter: => Boolean, addListener: (() => Unit) => Unit) extends CheckBox(text) {
  addListener(update)
  update()
  peer.addActionListener(new ActionListener() {
    def actionPerformed(ae: ActionEvent) = setter(peer.isSelected)
  });
  def update() = peer.setSelected(getter)
}

File: branches/test-branch-44224/simulations-java/simulations/density/src/edu/colorado/phet/densityjava/common/MyCheckBox.java
package edu.colorado.phet.densityjava.common;

import javax.swing.*;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class MyCheckBox extends JCheckBox {

    public MyCheckBox(String name, Model model, final Getter<Boolean> getter, final Setter<Boolean> setter) {
        super(name);
        setSelected(getter.getValue());
        addActionListener(new ActionListener() {
            public void actionPerformed(ActionEvent e) {
                setter.setValue(isSelected());
                setSelected(isSelected());
            }
        });
        model.addListener(new Unit() {
            public void update() {
                setSelected(getter.getValue());
            }
        });
    }
}